core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -776,6 +776,7 @@ private[spark] class TaskSetManager(
 
   /** Called by TaskScheduler when an executor is lost so we can re-enqueue our tasks */
   override def executorLost(execId: String, host: String, reason: ExecutorLossReason) {
+    if (isZombie) return
```

Comment:
@kayousterhout I agree with you, it's reasonable to add "!zombie" to the if-condition below, on line 784. 
@mccheah Sorry, I haven't kept up your mind. Will it be fine to just add "!zombie" to the if-condition below, on line 784, or there's a better way? Thanks.